### DISCUSSION TITLE
How to Publish / Subscribe MQTT messaging (mosquitto) dynamically?

### DISCUSSION THREAD

**Discussion by lcy-1024** [quarkusio/quarkus#26108] [Q&A]
This example is not dynamically:
https://github.com/quarkusio/quarkus-quickstarts/tree/main/mqtt-quickstart

I found the same question in stackoverflow:
https://stackoverflow.com/questions/62883516/publish-subscribe-mqtt-using-smallrye-reactive-messaging-dynamically

**Comment by ozangunalp** [comment]
In order to publish to different topics from the same channel, you can use the `MqttMessage.of(topic, ...)` signature to create your message with a specific topic. Then you can use emitter or outgoing channel to send that message to the broker.

For consuming events from multiple topics, the channel configuration `topic` supports `+` and `#`  wildcards. If different topics you'd like to consume don't fit the wildcards, you need to create multiple channels.

**Reply by lcy-1024** [reply]
Thank you for your reply.
Is there any way to consuming events without channel mode, just like publish topics we can use emitter instead of channel mode.

Is there a way to run without using the following configuration:
https://github.com/quarkusio/quarkus-quickstarts/blob/main/mqtt-quickstart/src/main/resources/application.properties
